2.sql_where.sql

```diff
@@ -0,0 +1,106 @@
+FROM Sales.SalesOrderHeader
+WHERE TotalDue BETWEEN 870 AND 1000
+and OrderDate BETWEEN '2002-01-01' AND'2002-06-30 '
+
+-- case 2: Importance of Paranthesis
+select Name, Color, ListPrice
+from Production.Product
+WHERE Color = 'White' OR Color = 'Gray' and ListPrice> 9
+
+--The above query will return everything that is white. In addition, the OR operator will also allow you to select all gray items,
+--but with a price greater than 9. Since we have not explicitly specified how individual conditions should be treated, 
+--all requirements that are connected with the AND operator will be treated together (obligatory). The others will be considered separately.
+-- The database engine processing the query treated the AND conditions together, so the previous entry without parentheses will be equivalent to the following:
+-- WHERE Color = 'White' OR (Color = 'Gray' and ListPrice> 9)
+
+select Name, Color, ListPrice
+from Production.Product
+WHERE (Color = 'White' OR Color = 'Gray') and ListPrice> 9
+
+-- case 3 : Filtering Character Values
+-- all records for which the value of the City column starts with "par"
+select distinct city
+from Person.Address
+where City LIKE 'par%'
+
+-- records for which the value of the City column contains the string "par" (plus any number of characters before and after)
+select distinct city
+from Person.Address
+where City LIKE '% par%' 
+
+--- all cities that start with any character,
+-- contain the sequence "par" in position 2-4 
+select distinct city
+from Person.Address
+where City LIKE '_par%'
+
+--Specifying a range of characters at the indicated position in square brackets []
+--The range can be continuous [az] or a list of possible [adg] characters. 
+--It can also specify the exclusion of specific characters - then we use the symbol ^. like : ring [td]
+select distinct city
+from Person.Address
+where City LIKE '% ring [td]%'  -- returns Springdale, Wamington
+
+select distinct city
+from Person.Address
+where City LIKE '% ring [^td]%' 
+
+--Scoping can be performed for several characters.
+--For example, the query below returns all items beginning with a or b and a second letter between n-p
+
+select distinct city
+from Person.Address
+where City LIKE '[ab] [np]%'  --Anacortes,Bobingy,Boise,Bonn,Boston,Bothell
+
+-- case 4 : Usage of where instead of join , the following queries are equal.
+-- Outer joins can be specified by using either the (+) syntax in the WHERE clause or the OUTER JOIN keywords in the FROM clause. 
+-- Note : The result of an outer join contains a copy of all rows from one table. In this topic, the table whose rows are preserved is called the “outer” table, and the other table is called the “inner” table.
+-- In a LEFT OUTER JOIN, the left-hand table is the outer table and the right-hand table is the inner table.
+-- In a RIGHT OUTER JOIN, the right-hand table is the outer table and the left-hand table is the inner table.
+
+
+--inner join : 
```

Comment:
It's not an inner join, it's a self join!

Reply:
Thanks for your comment but I don't think it is a self join as the t1 and t2 table are different.

Reply:
Please take a look at that url => https://www.w3schools.com/sql/sql_join_self.asp

Reply:
ok:/